soccer/src/soccer/strategy/agent/position/runner.cpp

```diff
@@ -0,0 +1,61 @@
+#include "runner.hpp"
+
+#include <spdlog/spdlog.h>
+
+namespace strategy {
+
+Runner::Runner(int r_id) : Position(r_id) {
+    position_name_ = "Runner";
+    current_state_ = LEFT_SIDE;
+}
+
+Runner::State Runner::update_state() {
+    State next_state = current_state_;
+    rj_geometry::Point robot_position = this->last_world_state_->get_robot(true, robot_id_).pose.position();
+
+    if (robot_position.nearly_equals(top_left_target_, 0.1)) {
```

Comment:
For simple PathTargetMotionCommands, this is a lot simpler and more accurate. Only use something like manually checking position if you can empirically show that the `check_is_done` is not an accurate condition for state transition. Even still, you should make it an OR statement between the two.

```suggestion
    if (check_is_done()) {
```